openstack/identity/v3/users/requests.go

```diff
@@ -0,0 +1,90 @@
+package users
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to
+// the List request
+type ListOptsBuilder interface {
+	ToUserListQuery() (string, error)
+}
+
+// ListOpts allows you to query the List method.
+type ListOpts struct {
+	// DomainID filters the response by a domain ID.
+	DomainID string `q:"domain_id"`
+
+	// Enabled filters the response by enabled users.
+	Enabled *bool `q:"enabled"`
+
+	// IdpID filters the response by an Identity Provider ID.
+	IdPID string `q:"enabled"`
+
+	// Name filters the response by username.
+	Name string `q:"name"`
+
+	// PasswordExpiresAt filters the response based on expiring passwords.
+	PasswordExpiresAt string `q:"password_expires_at"`
+
+	// ProtocolID filters the response by protocol ID.
+	ProtocolID string `q:"protocol_id"`
+
+	// UniqueID filters the response by unique ID.
+	UniqueID string `q:"unique_id"`
+}
+
+// ToUserListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToUserListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// List enumerates the Users to which the current token has access.
+func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(client)
+	if opts != nil {
+		query, err := opts.ToUserListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return UserPage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// GetOptsBuilder allows extensions to add additional parameters to
+// the Get request.
+type GetOptsBuilder interface {
+	ToUserGetQuery() (string, error)
+}
+
+// GetOpts allows you to modify the details included in the Get request.
+type GetOpts struct{}
```

Comment:
I agree. I did the same for Projects, but in hindsight, I have no idea why. I _must_ have had a reason or known of some customization in the wild.

I'm OK with removing this and adding it later if it comes up.